react/SellerWrapper.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback, useEffect, useRef } from 'react'
+import { useProduct, useProductDispatch } from 'vtex.product-context'
+import { useCssHandles } from 'vtex.css-handles'
+import type { Item } from 'vtex.product-context/react/ProductTypes'
+
+import useSeller from './hooks/useSeller'
+
+const CSS_HANDLES = ['sellerWrapper']
+const SELECT_ITEM_EVENT = 'SET_SELECTED_ITEM'
+
+type SellerWrapperProps = {
+  children: React.ReactNode
+}
+
+const SellerWrapper = ({ children }: SellerWrapperProps) => {
+  const { seller } = useSeller()
+  const dispatch = useProductDispatch()
+  const { selectedItem, product } = useProduct() ?? {}
+  const latestItem = useRef((null as unknown) as Item)
+  const handles = useCssHandles(CSS_HANDLES)
+
+  const memoizedCallback = useCallback(
+    itemSeller => ({
+      ...itemSeller,
+      sellerDefault: itemSeller.sellerId === seller,
+    }),
+    [seller]
+  )
+
+  const newCurrentSelectedItem = product?.items?.find(item =>
+    item.sellers?.find(
+      itemSeller =>
+        itemSeller.sellerId === seller &&
+        itemSeller.commertialOffer.AvailableQuantity > 0
+    )
+  )
+
+  useEffect(() => {
+    if (!seller || !newCurrentSelectedItem || !selectedItem || !dispatch) return
+
+    const { sellers } = newCurrentSelectedItem
+
+    dispatch?.({
+      type: SELECT_ITEM_EVENT,
+      args: {
+        item: {
+          ...newCurrentSelectedItem,
+          sellers: sellers.map(memoizedCallback),
+        },
+      },
+    })
+  }, [seller])
```

Comment:
This effect actually depends on more variables, at least `seller, newCurrentSelectedItem, selectedItem, dispatch, memoizedCallback`. Can you add them all? Ideally we'd have a lint rule to make sure that all the dependencies of an effect are listed, but it seems like it's off right now.

A way to reduce the number of dependencies here would be to have a separate variable for the first check done by this effect, for example:

```js
const shouldDispatchNewSelectedItem = seller && newCurrentSelectedItem && selectedItem && dispatch
```

That way you'll only need to list `shouldDispatchNewSelectedItem`, `seller` and `memoizedCallback` here.